### PR TITLE
DTSPB-1091-Modify-Caveat-Entered-Email

### PR DESCRIPTION
**Pull request by stelioskalo**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DTSPB-1091

### Change description ###
Modified CaveatPersonalisationService to push the date of death and birth to the email templates.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/probate/service/notification/CaveatPersonalisationService.java [hunk 1]
```diff
@@ -94,9 +100,31 @@
                 dateFormatterService.formatCaveatExpiryDate(caveatData.getExpiryDate()));
         personalisation.put(PERSONALISATION_WELSH_CAVEAT_EXPIRY_DATE,
                         localDateToWelshStringConverter.convert(caveatData.getExpiryDate()));
+        passDod(caveatData, personalisation);
+        checkIfDobExists(caveatData, personalisation);
 
         return personalisation;
     }
 
+    private void checkIfDobExists(CaveatData caveatData, HashMap<String, String> personalisation){
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/service/notification/CaveatPersonalisationService.java]: perhaps this method should be renamed - getDOBPersonalisation?

File: src/main/java/uk/gov/hmcts/probate/service/notification/CaveatPersonalisationService.java [hunk 2]
```diff
@@ -94,9 +100,31 @@
...
+            personalisation.put(PERSONALISATION_WELSH_DATE_OF_BIRTH,
+                    "Dyddiad marwolaeth yr ymadawedig yw: " + localDateToWelshStringConverter.convert(caveatData.getDeceasedDateOfBirth()));
+        }
+        else {
+            personalisation.put(PERSONALISATION_DATE_OF_BIRTH, "");
+            personalisation.put(PERSONALISATION_WELSH_DATE_OF_BIRTH, "");
+        }
+    }
+
+    private void passDod(CaveatData caveatData, HashMap<String, String> personalisation){
```
Comment (sanjayparekh-hmcts) [inline src/main/java/uk/gov/hmcts/probate/service/notification/CaveatPersonalisationService.java]: rename as getDODPersonalisation ?

**Review by sanjayparekh-hmcts — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/probate/service/notification/CaveatPersonalisationService.java (hunk 1 above)
Comment (stelioskalo) [reply src/main/java/uk/gov/hmcts/probate/service/notification/CaveatPersonalisationService.java]: Done.

File: src/main/java/uk/gov/hmcts/probate/service/notification/CaveatPersonalisationService.java (hunk 2 above)
Comment (stelioskalo) [reply src/main/java/uk/gov/hmcts/probate/service/notification/CaveatPersonalisationService.java]: removed method all together.

**Review by sanjayparekh-hmcts — APPROVED** [review]

File: dependency-check-suppressions.xml [hunk 3]
```diff
@@ -72,7 +72,8 @@
         <cve>CVE-2021-24122</cve>
         <cve>CVE-2021-25122</cve>
         <cve>CVE-2021-25329</cve>
-         <cve>CVE-2020-9484</cve>
+        <cve>CVE-2020-9484</cve>
+        <cve>CVE-2020-9484</cve>
```
Comment (isha2040) [inline dependency-check-suppressions.xml]: think is suppression is added twice

File: src/contractTest/resources/application.yml [hunk 4]
```diff
@@ -451,3 +451,4 @@ ld:
     key: ${LD_USER_KEY:35222af3-7867-44ea-a1a0-f8fs32backend_dummy}
     firstName: Probate
     lastName: Backend
+
```
Comment (isha2040) [inline src/contractTest/resources/application.yml:454]: can we remove this space and keep this file same like prod please

File: src/functionalTest/java/uk/gov/hmcts/probate/functional/util/FunctionalTestUtils.java [hunk 5]
```diff
@@ -305,4 +311,39 @@ public String addAttribute(String json, String attributeKey, String attributeVal
         return json.replaceAll("\"applicationID\": \"603\",",
             "\"applicationID\": \"603\",\"" + attributeKey + "\": \"" + attributeValue + "\",");
     }
+
+    public String formatDate(LocalDate dateToConvert) {
+        if (dateToConvert == null) {
+            return null;
+        }
+        DateFormat originalFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.ENGLISH);
+        DateFormat targetFormat = new SimpleDateFormat("dd MMMMM yyyy");
```
Comment (isha2040) [inline src/functionalTest/java/uk/gov/hmcts/probate/functional/util/FunctionalTestUtils.java:320]: is this date format correct ? new SimpleDateFormat("dd MMMMM yyyy");?

File: src/main/java/uk/gov/hmcts/probate/service/notification/CaveatPersonalisationService.java [hunk 6]
```diff
@@ -94,9 +99,23 @@
...
                         localDateToWelshStringConverter.convert(caveatData.getExpiryDate()));
+        personalisation.put(PERSONALISATION_DATE_OF_DEATH,
+                dateFormatterService.formatDate(caveatData.getDeceasedDateOfDeath()));
+        getDOBPersonalisation(caveatData, personalisation);
 
         return personalisation;
     }
 
+    private void getDOBPersonalisation(CaveatData caveatData, HashMap<String, String> personalisation) {
+        //Whole text being pushed to the email template as the template does
```
Comment (isha2040) [inline src/main/java/uk/gov/hmcts/probate/service/notification/CaveatPersonalisationService.java]: do we need these comments ? can we squeeze to short and combined one if absolutely required

File: src/functionalTest/java/uk/gov/hmcts/probate/functional/util/FunctionalTestUtils.java (hunk 5 above)
Comment (sanjayparekh-hmcts) [reply src/functionalTest/java/uk/gov/hmcts/probate/functional/util/FunctionalTestUtils.java:320]: correct for the test I need

File: dependency-check-suppressions.xml (hunk 3 above)
Comment (sanjayparekh-hmcts) [reply dependency-check-suppressions.xml]: removed

File: src/contractTest/resources/application.yml (hunk 4 above)
Comment (sanjayparekh-hmcts) [reply src/contractTest/resources/application.yml:454]: removed

File: src/main/java/uk/gov/hmcts/probate/service/notification/CaveatPersonalisationService.java (hunk 6 above)
Comment (sanjayparekh-hmcts) [reply src/main/java/uk/gov/hmcts/probate/service/notification/CaveatPersonalisationService.java]: removed

**Review by isha2040 — APPROVED** [review]

**Pull request merged**